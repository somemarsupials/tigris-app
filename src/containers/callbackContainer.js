import React, { Component } from 'react';
import { connect } from 'react-redux';
import { CallbackPage } from '../components/pages';
import callbackActions from '../actions/callback';

export class CallbackContainer extends Component {
  componentDidMount() {
    this.props.getToken(this.props.match.params.provider);
  }

  render() {
    if (this.props.success) {
      window.close();
    }

    return (
      <CallbackPage 
        error={this.props.failure} 
      />
    );
  }
}

export function mapStateToProps(state) {
  return {
    success: state.callback.success,
    failure: state.callback.failure,
  };
}

export function mapDispatchToState(dispatch) {
  return {
    getToken: provider => {
      dispatch(callbackActions.getToken(provider));
    },
  };
}

export default 
connect(mapStateToProps, mapDispatchToState)(CallbackContainer);
