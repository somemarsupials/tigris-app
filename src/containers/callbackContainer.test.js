import React from 'react';
import { shallow } from 'enzyme';
import { CallbackContainer } from './callbackContainer';

describe('CallbackContainer', () => {
  let props;

  beforeEach(() => {
    props = {
      match: {
        params: {
          provider: 'spotify',
        },
      },
      getToken: jest.fn(),
    };
    global.window.close = jest.fn();
  });

  describe('on mount', () => {
    beforeEach(() => {
      shallow(<CallbackContainer {...props} />);
    });

    it('calls getToken with provider', () => {
      expect(props.getToken).toHaveBeenCalledWith('spotify');
    });
  });

  describe('when rendering', () => {
    describe('when successful', () => {
      beforeEach(() => {
        props.success = true;
        shallow(<CallbackContainer {...props} />);
      });

      it('closes window', () => {
        expect(global.window.close).toHaveBeenCalled();
      });
    });

    describe('when unsuccessful', () => {
      describe('with no error', () => {
        it('matches snapshot', () => {
          expect(shallow(<CallbackContainer {...props} />))
            .toMatchSnapshot();
        });
      });

      describe('with error', () => {
        beforeEach(() => {
          props.error = {
          };
        });

        it('matches snapshot', () => {
          expect(shallow(<CallbackContainer {...props} />))
            .toMatchSnapshot();
        });
      });
    });
  });
});
