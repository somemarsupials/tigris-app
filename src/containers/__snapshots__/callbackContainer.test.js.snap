// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallbackContainer when rendering when unsuccessful with error matches snapshot 1`] = `<_default />`;

exports[`CallbackContainer when rendering when unsuccessful with no error matches snapshot 1`] = `<_default />`;
